DSL/transform.md

```diff
@@ -0,0 +1,364 @@
+# Transformation DSL
+
+This DSL (Domain-Specific Language) is inspired by JSON Patch and follows RFC 6901 for JSON Pointer representation in the path. It consists of an array of objects, each having the following keys:
+
+## Tranform fields
+
+| Property  | Description                                                      | Required | Data Type         |
+|-----------|------------------------------------------------------------------|----------|-------------------|
+| operation | Specifies the operation to perform                               | Yes      | String            |
+| path      | Specifies the path to the target in the schema                   | Yes      | Array of Strings  |
+| value     | Specifies the value to be used in the operation                  | No       | Any               |
+| from      | Specifies the source path for operations involving move/copy     | No       | Array of Strings  |
+| to        | Specifies the destination path for operations involving move/copy| No       | Array of Strings  |
+
+## Operations
+
+### 1. `append-property`
+
+Appends property in the provided `from` path to the `to` path. The `to` path should be an array, or it will be created as an empty array before appending.
+
+- **Takes**: `from`, `to`
```

Comment:
Can you turn these into tables with a short description for each argument?

Reply:
Also, most other operations use `path` and `value`. Can it be the same here?

Reply:
I think no, because `value`: something, will take direct value(whatever is given in the value). So, if we change the value to have array of anything it will simply mean we are saying to append array, or `value: {}` will mean to append empty object.
Am I right?

Reply:
Hmm I think we can at least rename `from` to `path` (because it's actually a path?)

Reply:
> Hmm I think we can at least rename `from` to `path` (because it's actually a path?)

Consider we are using `add` operation. `{ op: add, path: [ new ], value: 123}`. Here the path is destination location, so it causes confusion....
Even json-patch uses from-to, path-value combinations

Reply:
Alright. Sounds good!